jax/_src/scipy/optimize/line_search.py

```diff
@@ -384,25 +386,28 @@ def body(state):
                      state)
 
   status = jnp.where(
-      state.failed & (~state.saddle_point),
+      state.failed,
       jnp.array(1),  # zoom failed
-      jnp.where(
-          state.failed & state.saddle_point,
-          jnp.array(2),  # saddle point reached,
           jnp.where(
               state.i > maxiter,
               jnp.array(3),  # maxiter reached
               jnp.array(0),  # passed (should be)
           ),
-      ),
   )
+  # Step sizes which are too small causes the optimizer to get stuck with a
+  # direction of zero in  <64 bit mode - avoid this with  a minimum step size.
+  alpha_k = state.a_star
+  if not jax.config.x64_enabled:
```

Comment:
Is there any way we can detect this based on the `dtype` of some arrays? Or perhaps we can just do it in float64 mode too (with a much smaller threshold)?

We'd really like JAX to treat float32 values in a fully consistent way even if x64 mode is enabled, at least if the function is carefully written to only use float32 numbers internally and all the inputs are float32.

Even if that isn't true of the current `jax.scipy.optimize.minimize` routines, I would prefer not to introduce this explicit dependency on `jax.config`, which would be a step in the wrong direction.

Reply:
That makes sense - I've updated it to be dependent on the type of the array.

I used the number of bits from `jnp.finfo` so that it would also work for bfloat16, etc.

Happy to change that if there's a more standard way of checking the precision of `dtype` for arrays - I couldn't find an obvious helper but may have been looking in the wrong place.